PowerShell Scanners/Cipher Suite Detection/Cipher Suite Detection.xml

```diff
@@ -0,0 +1,116 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<AdminArsenal.Export Code="PDQInventory" Name="PDQ Inventory" Version="19.1.96.0" MinimumVersion="19.0">
+  <ScanProfile>
+    <Collections type="list" />
+    <Scanners type="list">
+      <Scanner>
+        <ModifiedDate>2021-01-06T17:07:06.0000000-05:00</ModifiedDate>
+        <Name>detect_ciphers</Name>
+        <UID>b4d6c161208d4501955bb66b5641d436</UID>
+        <Script>#Query Registry for cipher suites
+
+#SSLv2
+$SSLv2Server = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\SSL 2.0\Server' -Name 'Enabled'
+$SSLv2Client = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\SSL 2.0\Client' -Name 'Enabled'
+
+#SSLv3
+$SSLv3Server = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\SSL 3.0\Server' -Name 'Enabled'
+$SSLv3Client = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\SSL 3.0\Client' -Name 'Enabled'
+
+#TLSv1.0
+$TLSv1Server = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.0\Server' -Name 'Enabled'
+$TLSv1Client = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.0\Client' -Name 'Enabled'
+
+#TLSv1.1
+$TLSv1_1Server = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.1\Server' -Name 'Enabled'
+$TLSv1_1Client = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.1\Client' -Name 'Enabled'
```

Comment:
As written, if any of these registry keys don't exist on the target, the outcome of your IF statements automatically becomes $true - your IF statements **require** it to equal 0 in order for $false to be set. Anything else passes. With lower versions of the protocols Windows has them disabled by default (so no key exists yet).
![image](https://user-images.githubusercontent.com/20692061/103942782-29e14600-50ee-11eb-9bfb-df7899f2c23f.png)

You'll need to add in some additional error checking to validate the results before we can merge. 

Here's a quick example I threw together to give an idea of one way to handle it. I didn't have time to test it out, but the overall approach would work. 
In this situation, there is only 1 scenario where SSLv2 would be enabled (the key exists), so I used that as the validation:

```powershell
$SSLv2Server = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\SSL 2.0\Server"

if (Test-Path -Path $SSLv2Server) {
    if ((Get-ItemPropertyValue -Path $SSLv2Server -Name Enabled -ErrorAction SilentlyContinue) -eq "1") {
        $SSLv2ServerEnabled = $true
    }
}
else {
    $SSLv2ServerEnabled = $false
}

Reply:
I just commited changes to address this issue.